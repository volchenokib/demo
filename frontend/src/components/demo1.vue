<template>
  <v-container>
    <v-layout column>
      <v-card class="pa-4 my-0 mx-auto dropDown" max-width="650" elevation="8">
        <v-form v-model="demoForm" ref="form" name="demoForm" class="text-xs-center">
          <v-layout row wrap justify-space-between>
            <!-- picker #1 -->
            <v-flex xs12 sm5>
              <v-menu
                ref="menuFrom"
                v-model="menuFrom"
                :close-on-content-click="false"
                lazy
                transition="scale-transition"
                max-width="290px"
                min-width="290px"
                offset-y
                full-width
                :nudge-right="0"
                :nudge-bottom="30"
              >
                <template v-slot:activator="{ on }">
                  <v-text-field
                    v-model="computedDateFrom"
                    label="от"
                    persistent-hint
                    prepend-icon="event"
                    readonly
                    required
                    :rules="fromDateRules"
                    v-on="on"
                    color="primary"
                  ></v-text-field>
                </template>

                <v-date-picker
                  v-model="dateFrom"
                  min="2014-01-01"
                  :max="dateTo || today"
                  no-title
                  @input="menuFrom = false"
                  scrollable
                  first-day-of-week="1"
                  locale="ru-RU"
                  color="primary"
                ></v-date-picker>
              </v-menu>
            </v-flex>
            <!-- picker #2 -->
            <v-flex xs12 sm5>
              <v-menu
                v-model="menuTo"
                :close-on-content-click="false"
                lazy
                transition="scale-transition"
                max-width="290px"
                min-width="290px"
                offset-y
                full-width
                :nudge-right="0"
                :nudge-bottom="30"
              >
                <template v-slot:activator="{ on }">
                  <v-text-field
                    v-model="computedDateTo"
                    label="до"
                    persistent-hint
                    prepend-icon="event"
                    readonly
                    required
                    :rules="toDateRules"
                    v-on="on"
                    color="primary"
                  ></v-text-field>
                </template>

                <v-date-picker
                  v-model="dateTo"
                  :min="dateFrom || '2014-01-01'"
                  :max="today"
                  no-title
                  @input="menuTo = false"
                  scrollable
                  first-day-of-week="1"
                  locale="ru-RU"
                  color="primary"
                ></v-date-picker>
              </v-menu>
            </v-flex>
          </v-layout>

          <!-- radio buttons -->
          <v-radio-group v-model="purchaseType" :mandatory="true">
            <v-radio
              name="singleProvider"
              label="закупка у единственного поставщика"
              value="singleProvider"
              color="primary"
            ></v-radio>
            <v-radio name="eAuction" label="электронный аукцион" value="eAuction" color="primary"></v-radio>
          </v-radio-group>

          <!-- switcher -->
          <v-switch
            name="localStorage"
            :value="true"
            v-model="localStorage"
            label="Взять из локального хранилища"
            color="primary"
          ></v-switch>

          <v-divider></v-divider>

          <!-- button -->
          <v-btn
            ref="downloadButton"
            color="primary"
            flat
            :disabled="buttonDisable"
            :loading="dataLoading"
            @click="submit"
          >
            сформировать
            <template v-slot:loader>
              <span class="d-flex mt-2">
                <v-icon class="loader mr-1" light>cached</v-icon>Загрузка
              </span>
            </template>
          </v-btn>
        </v-form>
      </v-card>
    </v-layout>
  </v-container>
</template>

<style lang="scss">
.loader {
  animation: loader 1s infinite;
  margin-top: -4px;
}
@keyframes loader {
  from {
    transform: rotate(0);
  }
  to {
    transform: rotate(360deg);
  }
}
</style>

<script>
// TODO:
// + formData
// - remove getters
// - error handler
// - notifications
export default {
  name: "demo1",
  data: () => ({
    demoForm: "",
    fromDateRules: [v => !!v || "Это обязательное поле"],
    toDateRules: [v => !!v || "Это обязательное поле"],
    formattedDateFrom: null,
    dateFrom: null,
    formattedDateTo: null,
    dateTo: null,
    menuFrom: false,
    menuTo: false,
    purchaseType: "singleProvider",
    localStorage: false
  }),

  computed: {
    resetForm() {
      return this.$store.getters.formResetNeed;
    },
    dataLoading() {
      return this.$store.getters.getDataState;
    },
    buttonDisable() {
      return this.$store.getters.getButtonState;
    },
    computedDateFrom: {
      set: function() {
        this.dateFrom;
      },
      get: function() {
        if (this.dateFrom) {
          const [year, month, day] = this.dateFrom.split("-");
          return (this.formattedDateFrom = `${day}.${month}.${year}`);
        }
      }
    },
    computedDateTo: {
      set: function() {
        this.dateTo;
      },
      get: function() {
        if (this.dateTo) {
          const [year, month, day] = this.dateTo.split("-");
          return (this.formattedDateTo = `${day}.${month}.${year}`);
        }
      }
    },
    today() {
      var today = new Date();
      var dd = String(today.getDate()).padStart(2, "0");
      var mm = String(today.getMonth() + 1).padStart(2, "0"); //January is 0!
      var yyyy = today.getFullYear();

      return (today = yyyy + "-" + mm + "-" + dd);
    }
  },

  watch: {
    resetForm() {
      this.$refs.form.reset();
      this.dateFrom = null;
      this.dateTo = null;
      this.purchaseType = "singleProvider";
      this.localStorage = false;
      // this.$store.state.form.isReset = false;
      this.$store.commit("API_DATA_SUCCES", false);
    }
  },

  methods: {
    submit() {
      if (this.$refs.form.validate()) {
        const payload = new FormData(demoForm);

        payload.append("dateFrom", this.formattedDateFrom);
        payload.append("dateTo", this.formattedDateTo);

        // Display the keys
        // for (let key of payload.keys()) {
        //   console.log(key);
        // }

        // Display the values
        // for (let value of payload.values()) {
        //   console.log(value);
        // }

        this.$store
          .dispatch("getFile", payload)
          .then(response => this.download(response));
      }
    },

    download(response) {
      const url = window.URL.createObjectURL(new Blob([response.data]));
      const link = document.createElement("a");
      link.href = url;
      link.setAttribute("download", "file.xlsx");
      document.body.appendChild(link);
      link.click();
    }
  }
};
</script>
