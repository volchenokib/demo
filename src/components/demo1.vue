<template>
  <v-container>
    <v-layout column>
      <v-card class="pa-4 my-0 mx-auto dropDown" max-width="650" elevation="8">
        <v-form v-model="demoForm" ref="form" class="text-xs-center">
          <v-layout row wrap justify-space-between>
            <!-- picker #1 -->
            <v-flex xs12 sm5>
              <v-menu
                ref="menuFrom"
                v-model="menuFrom"
                :close-on-content-click="false"
                lazy
                transition="scale-transition"
                max-width="290px"
                min-width="290px"
                offset-y
                full-width
                :nudge-right="0"
                :nudge-bottom="30"
              >
                <template v-slot:activator="{ on }">
                  <v-text-field
                    v-model="computedDateFrom"
                    label="от"
                    persistent-hint
                    prepend-icon="event"
                    readonly
                    required
                    :rules="fromDateRules"
                    v-on="on"
                    color="primary"
                  ></v-text-field>
                </template>

                <v-date-picker
                  v-model="dateFrom"
                  min="2014-01-01"
                  :max="dateTo || today"
                  no-title
                  @input="menuFrom = false"
                  scrollable
                  first-day-of-week="1"
                  locale="ru-RU"
                  color="primary"
                ></v-date-picker>
              </v-menu>
            </v-flex>
            <!-- picker #2 -->
            <v-flex xs12 sm5>
              <v-menu
                v-model="menuTo"
                :close-on-content-click="false"
                lazy
                transition="scale-transition"
                max-width="290px"
                min-width="290px"
                offset-y
                full-width
                :nudge-right="0"
                :nudge-bottom="30"
              >
                <template v-slot:activator="{ on }">
                  <v-text-field
                    v-model="computedDateTo"
                    label="до"
                    persistent-hint
                    prepend-icon="event"
                    readonly
                    required
                    :rules="toDateRules"
                    v-on="on"
                    color="primary"
                  ></v-text-field>
                </template>

                <v-date-picker
                  v-model="dateTo"
                  :min="dateFrom || '2014-01-01'"
                  :max="today"
                  no-title
                  @input="menuTo = false"
                  scrollable
                  first-day-of-week="1"
                  locale="ru-RU"
                  color="primary"
                ></v-date-picker>
              </v-menu>
            </v-flex>
          </v-layout>

          <!-- radio buttons -->
          <v-radio-group v-model="purchaseType" :mandatory="true">
            <v-radio
              label="закупка у единственного поставщика"
              value="singleProvider"
              color="primary"
            ></v-radio>
            <v-radio label="электронный аукцион" value="eAuction" color="primary"></v-radio>
          </v-radio-group>

          <!-- switcher -->
          <v-switch v-model="localStorage" label="Взять из локального хранилища" color="primary"></v-switch>

          <v-divider></v-divider>

          <!-- button -->
          <v-btn
            color="primary"
            flat
            :disabled="buttonDisable"
            :loading="dataLoading"
            @click="download"
          >
            сформировать
            <template v-slot:loader>
              <span class="d-flex mt-2">
                <v-icon class="loader mr-1" light>cached</v-icon>Загрузка
              </span>
            </template>
          </v-btn>
        </v-form>
      </v-card>
    </v-layout>
  </v-container>
</template>

<style lang="scss">
.loader {
  animation: loader 1s infinite;
  margin-top: -4px;
}
@keyframes loader {
  from {
    transform: rotate(0);
  }
  to {
    transform: rotate(360deg);
  }
}
</style>


<script>
export default {
  name: "demo1",
  data: () => ({
    demoForm: "",
    fromDateRules: [v => !!v || "Это обязательное поле"],
    toDateRules: [v => !!v || "Это обязательное поле"],
    formattedDateFrom: null,
    dateFrom: null,
    formattedDateTo: null,
    dateTo: null,
    menuFrom: false,
    menuTo: false,
    purchaseType: "singleProvider",
    localStorage: false
  }),

  computed: {
    resetForm() {
      return this.$store.getters.formResetNeed;
    },
    dataLoading() {
      return this.$store.getters.getDataState;
    },
    buttonDisable() {
      return this.$store.getters.getButtonState;
    },
    computedDateFrom: {
      set: function() {
        this.dateFrom;
      },
      get: function() {
        console.log("dateFrom", this.dateFrom);
        if (this.dateFrom) {
          const [year, month, day] = this.dateFrom.split("-");
          return (this.formattedDateFrom = `${day}.${month}.${year}`);
        }
      }
    },
    computedDateTo: {
      set: function() {
        this.dateTo;
      },
      get: function() {
        if (this.dateTo) {
          const [year, month, day] = this.dateTo.split("-");
          return (this.formattedDateTo = `${day}.${month}.${year}`);
        }
      }
    },
    today() {
      var today = new Date();
      var dd = String(today.getDate()).padStart(2, "0");
      var mm = String(today.getMonth() + 1).padStart(2, "0"); //January is 0!
      var yyyy = today.getFullYear();

      return (today = yyyy + "-" + mm + "-" + dd);
    }
  },

  watch: {
    resetForm() {
      this.$refs.form.reset();
      this.dateFrom = null;
      this.dateTo = null;
      this.purchaseType = "singleProvider";
      this.localStorage = false;
      // this.$store.state.form.isReset = false;
      this.$store.commit("API_DATA_SUCCES", false);
      console.log("watcher", this.$store.state.form.isReset);
    }
  },

  methods: {
    download() {
      if (this.$refs.form.validate()) {
        const payload = {
          dateFrom: this.formattedDateFrom,
          dateTo: this.formattedDateTo,
          purchaseType: this.purchaseType,
          localStorage: this.localStorage
        };

        this.$store.dispatch("downloadFile", payload);
        console.log("payload", payload);
      }
    }
  }
};
</script>
