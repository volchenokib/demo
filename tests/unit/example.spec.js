import Vue from 'vue';
import Vuex from 'vuex';
import Vuetify from 'vuetify';
import { shallowMount, mount, createLocalVue } from '@vue/test-utils';
import mockAxios from 'axios';
import demo1 from '@/components/demo1.vue';
import downloadFile from '@/api/demo.js';

Vue.use(Vuetify);
Vue.use(Vuex);

let getters;
let actions;
let store;

store = new Vuex.Store({
	getters,
	actions
});

describe('demo1.vue', () => {
	const wrapper = mount(demo1, { store });

	it('renders correctly', () => {
		expect(wrapper.element).toMatchSnapshot();
	});

	it('file download', async () => {
		// setup
		mockAxios.get.mockImplementationOnce(() =>
			Promise.resolve({
				data: { results: ['cat.jpg'] }
			})
		);

		// work
		const images = await downloadFile('cats');

		// expect
		expect(images).toEqual(['cat.jpg']);
		expect(mockAxios.get).toHaveBeenCalledTimes(1);
		expect(mockAxios.get).toHaveBeenCalledWith(
			'https://78.media.tumblr.com/tumblr_m39nv7PcCU1r326q7o1_500.png',
			{
				// params: {
				// 	client_id: process.env.REACT_APP_UNSPLASH_TOKEN,
				// 	query: "cats"
				// }
			}
		);
	});

	// it('submit button works correctly', () => {
	// 	const input = wrapper.findAll('input[type="text"]');
	// 	input.setValue('10.04.2019');
	// 	const button = wrapper.find('button');
	// 	button.trigger('click');

	// 	// assert event has been emitted
	// 	expect(wrapper.emitted().download).toBeTruthy();

	// 	// assert event count
	// 	expect(wrapper.emitted().download.length).toBe(1);
	// });
});
