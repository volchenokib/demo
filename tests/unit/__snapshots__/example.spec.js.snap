// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`demo1.vue renders correctly 1`] = `
<div
  class="container"
>
  <div
    class="layout column"
  >
    <div
      class="pa-4 my-0 mx-auto dropDown v-card v-sheet theme--light elevation-8"
      style="max-width: 650px;"
    >
      <form
        class="v-form text-xs-center"
        novalidate="novalidate"
      >
        <div
          class="layout row wrap justify-space-between"
        >
          <div
            class="flex xs12 sm5"
          >
            <div
              class="v-input v-text-field v-input--is-readonly theme--light"
            >
              <div
                class="v-input__prepend-outer"
              >
                <div
                  class="v-input__icon v-input__icon--prepend"
                >
                  <i
                    aria-hidden="true"
                    class="v-icon material-icons theme--light"
                  >
                    event
                  </i>
                </div>
              </div>
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <div
                    class="v-text-field__slot"
                  >
                    <label
                      aria-hidden="true"
                      class="v-label theme--light"
                      style="left: 0px; position: absolute;"
                    >
                      от
                    </label>
                    <input
                      aria-label="от"
                      readonly="readonly"
                      required="required"
                      type="text"
                    />
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <div
                      class="v-messages__wrapper"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="v-menu"
            >
              <div
                class="v-menu__content theme--light "
                style="max-height: auto; min-width: 290px; max-width: 290px; top: 30px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
              />
            </div>
          </div>
           
          <div
            class="flex xs12 sm5"
          >
            <div
              class="v-input v-text-field v-input--is-readonly theme--light"
            >
              <div
                class="v-input__prepend-outer"
              >
                <div
                  class="v-input__icon v-input__icon--prepend"
                >
                  <i
                    aria-hidden="true"
                    class="v-icon material-icons theme--light"
                  >
                    event
                  </i>
                </div>
              </div>
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <div
                    class="v-text-field__slot"
                  >
                    <label
                      aria-hidden="true"
                      class="v-label theme--light"
                      style="left: 0px; position: absolute;"
                    >
                      до
                    </label>
                    <input
                      aria-label="до"
                      readonly="readonly"
                      required="required"
                      type="text"
                    />
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <div
                      class="v-messages__wrapper"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="v-menu"
            >
              <div
                class="v-menu__content theme--light "
                style="max-height: auto; min-width: 290px; max-width: 290px; top: 30px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
              />
            </div>
          </div>
        </div>
         
        <div
          class="v-input v-input--selection-controls v-input--radio-group v-input--radio-group--column v-input--is-label-active v-input--is-dirty theme--light"
        >
          <div
            class="v-input__control"
          >
            <div
              class="v-input__slot"
              style="height: auto;"
            >
              <div
                class="v-input--radio-group__input"
                role="radiogroup"
              >
                <div
                  class="v-radio theme--light primary--text"
                >
                  <div
                    class="v-input--selection-controls__input"
                  >
                    <input
                      aria-checked="true"
                      aria-label="закупка у единственного поставщика"
                      name="v-radio-21"
                      role="radio"
                      type="radio"
                      value="singleProvider"
                    />
                    <div
                      class="v-input--selection-controls__ripple primary--text"
                    />
                    <i
                      aria-hidden="true"
                      class="v-icon material-icons theme--light primary--text"
                    >
                      radio_button_checked
                    </i>
                  </div>
                  <label
                    aria-hidden="true"
                    class="v-label theme--light"
                    style="left: 0px; position: relative;"
                  >
                    закупка у единственного поставщика
                  </label>
                </div>
                 
                <div
                  class="v-radio theme--light"
                >
                  <div
                    class="v-input--selection-controls__input"
                  >
                    <input
                      aria-checked="false"
                      aria-label="электронный аукцион"
                      name="v-radio-21"
                      role="radio"
                      type="radio"
                      value="eAuction"
                    />
                    <div
                      class="v-input--selection-controls__ripple"
                    />
                    <i
                      aria-hidden="true"
                      class="v-icon material-icons theme--light"
                    >
                      radio_button_unchecked
                    </i>
                  </div>
                  <label
                    aria-hidden="true"
                    class="v-label theme--light"
                    style="left: 0px; position: relative;"
                  >
                    электронный аукцион
                  </label>
                </div>
              </div>
            </div>
            <div
              class="v-messages theme--light"
            >
              <span
                class="v-messages__wrapper"
                name="message-transition"
                tag="div"
              />
            </div>
          </div>
        </div>
         
        <div
          class="v-input v-input--selection-controls v-input--switch theme--light"
        >
          <div
            class="v-input__control"
          >
            <div
              class="v-input__slot"
            >
              <div
                class="v-input--selection-controls__input"
              >
                <input
                  aria-checked="false"
                  aria-label="Взять из локального хранилища"
                  role="checkbox"
                  type="checkbox"
                  value=""
                />
                <div
                  class="v-input--selection-controls__ripple"
                />
                <div
                  class="v-input--switch__track theme--light"
                />
                <div
                  class="v-input--switch__thumb theme--light"
                >
                  <!---->
                </div>
              </div>
              <label
                aria-hidden="true"
                class="v-label theme--light"
                style="left: 0px; position: relative;"
              >
                Взять из локального хранилища
              </label>
            </div>
            <div
              class="v-messages theme--light"
            >
              <span
                class="v-messages__wrapper"
                name="message-transition"
                tag="div"
              />
            </div>
          </div>
        </div>
         
        <hr
          class="v-divider theme--light"
        />
         
        <button
          class="v-btn v-btn--flat theme--light primary--text"
          type="button"
        >
          <div
            class="v-btn__content"
          >
            
          сформировать
          
          </div>
        </button>
      </form>
    </div>
  </div>
</div>
`;
